plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/model/UpdateCenterData.java

```diff
@@ -0,0 +1,42 @@
+package io.jenkins.tools.pluginmodernizer.core.model;
+
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class UpdateCenterData {
+    private final JsonNode jsonNode;
+
+    public UpdateCenterData(JsonNode jsonNode) {
+        this.jsonNode = Objects.requireNonNull(jsonNode, "jsonNode must not be null");
+    }
+
+    public JsonNode getPlugin(String pluginName) {
```

Comment:
I think we really miss a Plugin class POJO (or record) to return such info

```java
public Plugin getPlugin(String pluginName) {
```

Would be more elegant that exposing implementation detail

For me this code can be cleanup in an other PR